### PR TITLE
[test] feat: add --force flag to okteto context use command

### PR DESCRIPTION
**Pull request by rberrelleza**
## Overview

This PR adds a new `--force` parameter to the `okteto context use` command that deletes the corresponding configuration in the okteto context and forces the user to log in again.

**Developed in collaboration with Okteto AI Agent Fleets** 🤖

### Core Implementation
- **Added `--force` flag** to `okteto context use` command
- **Context deletion logic** that handles both with/without schema formats
- **Token clearing** to ensure fresh authentication
- **Current context cleanup** when deleting active context

### Files Modified
- `cmd/context/options.go` - Added Force field to Options struct
- `cmd/context/use.go` - Added --force flag definition
- `cmd/context/create.go` - Implemented force handling logic
- `cmd/context/use_test.go` - Added comprehensive unit tests

### Documentation
- `IMPLEMENTATION_SUMMARY.md` - Detailed implementation overview
- `MANUAL_TEST.md` - Manual testing guide with scenarios

## Functionality

When `--force` flag is used:
1. 🗑️ Deletes existing context configuration from context store
2. 🔑 Clears cached authentication tokens
3. 🔄 Forces re-authentication on next use
4. 📝 Handles both URL formats (with/without https schema)
5. 🧹 Cleans up current context if it matches deleted context

## Usage

```bash
# Force re-authentication for a context
okteto context use https://okteto.example.com --force

# Works with or without schema
okteto context use okteto.example.com --force
```

### Unit Tests ✅
- 6 comprehensive test scenarios covering:
  - Force flag functionality
  - Context deletion with different URL formats
  - Token clearing behavior
  - Edge cases (non-existent contexts)
  - Flag definition validation

### Manual Testing Guide ✅
- 4 different test scenarios provided
- Step-by-step verification commands
- Expected behavior documentation

## Backward Compatibility ✅

- **No breaking changes** - flag defaults to `false`
- **Existing functionality preserved** when flag not used
- **Consistent with existing patterns** in the codebase

## Security Considerations ✅

- **Token clearing** ensures no cached credentials remain
- **Context isolation** - only affects specified context
- **Fresh authentication** required after force operation

## Checklist

- [x] Implementation complete
- [x] Unit tests added and passing
- [x] Manual testing guide provided
- [x] Documentation included
- [x] Backward compatibility maintained
- [x] Security considerations addressed

---

*This feature was developed in collaboration with Okteto AI Agent Fleets to enhance the user experience when managing Okteto contexts.*

### CODE REVIEW

**Comment by rberrelleza** [comment]
**Related Ticket**: https://okteto.atlassian.net/browse/PROD-361

File: cmd/context/use.go [hunk 1]
```diff
@@ -137,7 +164,17 @@ func (c *Command) Run(ctx context.Context, ctxOptions *Options) error {
 		ctxOptions.Save = true
 	}
 
+	if err := c.forceLoginIfRequested(ctxOptions, ctxStore); err != nil {
+		return err
+	}
+
 	if err := c.UseContext(ctx, ctxOptions); err != nil {
+		// delete the context to force the user to log in again
+		if oktetoErrors.IsNotFound(err) {
```
Comment (rberrelleza) [inline cmd/context/use.go:187]: this is handling the specific error we saw on the replicated migration:
```
 x  Non-200 OK status code: 401 Unauthorized body: "fail to find user with token XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXx309: not-found\n"
```

This happens when you don't have a valid token on your context file. I think we should fix this at the root, but I don't want the agent to do that big of a refactoring.  Without looking too deep, I think this could be handled on https://github.com/okteto/okteto/blob/master/cmd/context/create.go#L451 instead

**Comment by jLopezbarb** [comment]
Hi @rberrelleza, I think #4747 covers the main issue that for what this was created. I think adding a mechanism for doing a re-login could be useful for some scenarios, but we need to think about the experience. I don't think `force` would be the right naming for this. Let's discuss offline what would be the best option for this and then reopen it. WDYT?
